From 3715df1fa0faf30139cf7cdf67f98863aea52633 Mon Sep 17 00:00:00 2001
From: Li Wang <li.wang@windriver.com>
Date: Thu, 17 Dec 2020 05:12:58 +0000
Subject: [PATCH] zynqmp/clkc: fix -Wstringop-truncation warnings

Replace strncpy() with strscpy(), fixes the following warning:
drivers/clk/zynqmp/clkc.c:158:3: warning: 'strncpy' specified bound 50 equals destination size [-Wstringop-truncation]

Signed-off-by: Li Wang <li.wang@windriver.com>
---
 drivers/clk/zynqmp/clkc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/zynqmp/clkc.c b/drivers/clk/zynqmp/clkc.c
index 3209b679f1fe..f8875bacf616 100644
--- a/drivers/clk/zynqmp/clkc.c
+++ b/drivers/clk/zynqmp/clkc.c
@@ -155,7 +155,7 @@ static int zynqmp_get_clock_name(u32 clk_id, char *clk_name)
 
 	ret = is_valid_clock(clk_id, &valid);
 	if (!ret && valid) {
-		strncpy(clk_name, clock[clk_id].clk_name, MAX_NAME_LEN);
+		strscpy(clk_name, clock[clk_id].clk_name, MAX_NAME_LEN);
 		return 0;
 	} else {
 		return ret;
-- 
2.26.1

